.github/workflows/promote_charm.yaml

```diff
@@ -20,10 +24,23 @@ jobs:
     needs: get-runner-image
     steps:
       - uses: actions/checkout@v3
+      - name: Get build-on value
+        run: |
+          name="ubuntu"
+          channel="20.04"
+          if [ -f charmcraft.yaml ]; then
```

Comment:
Have you considered moving this logic to the `get_runner_image.yaml` file? Maybe we can parse the charmcraft.yaml file there and get all the needed options.

Reply:
This is a good idea. I updated the code.